src/ADAL.PCL/AdalHttpClient.cs

```diff
@@ -94,6 +103,21 @@ public AdalHttpClient(string uri, CallState callState)
                         string[] errorCodes = tokenResponse.ErrorCodes ?? new[] {ex.WebResponse.StatusCode.ToString()};
                         serviceEx = new AdalServiceException(tokenResponse.Error, tokenResponse.ErrorDescription,
                             errorCodes, ex);
+
+                        if ((ex.WebResponse.StatusCode.Equals(HttpStatusCode.InternalServerError)) ||
+                            (ex.WebResponse.StatusCode).Equals(HttpStatusCode.GatewayTimeout) ||
+                            (ex.WebResponse.StatusCode).Equals(HttpStatusCode.ServiceUnavailable))
+                        {
+                            if (RetryOnce)
+                            {
+                                await Task.Delay(DelayTimePeriod);
+                                RetryOnce = false;
+                                PlatformPlugin.Logger.Information(this.CallState, "WebResponse is not a success due to :-" + ex.InnerException + "Retrying one more time..");
```

Comment:
Consider using acronym instead of "is not success". this log message would not look formatted as innerException is being logged.